### ISSUE TITLE
Error: EPERM: operation not permitted, scandir

### ISSUE THREAD

**Issue by jbronikowski** [npm/npm#19361]
#### I'm opening this issue because:

  - [ ] npm is crashing.
  - [ ] npm is producing an incorrect install.
  - [ ] npm is doing something I don't understand.
  - [ ] Other (_see below for feature requests_):

### supporting information:

 - `npm -v` prints:
 - `node -v` prints:
 - `npm config get registry` prints:
 - Windows, OS X/macOS, or Linux?:
 - Network issues:
   - Geographic location where npm was run:
   - [ ] I use a proxy to connect to the npm registry.
   - [ ] I use a proxy to connect to the web.
   - [ ] I use a proxy when downloading Git repos.
   - [ ] I access the npm registry via a VPN
   - [ ] I don't use a proxy, but have limited or unreliable internet access.
 - Container:
   - [ ] I develop using Vagrant on Windows.
   - [ ] I develop using Vagrant on OS X or Linux.
   - [ ] I develop / deploy using Docker.
   - [ ] I deploy to a PaaS (Triton, Heroku).

**Comment by jbronikowski** [comment]
I was able to fix this by changing the permission of the folder i was trying to run npm install or ng new from to Everyone with full control

**Comment by kenany** [comment]
Closing as resolved.

**Comment by EBMike** [comment]
I just updated npm from 4.3 to 5.6 and now I have this problem.

@KenanY I don't see how changing a folders permissions to "Everyone" with full perms is deemed "resolved" (or for that matter, a parent folder of node_modules)

 Considering npm is the one creating the node_module folder in the first place this means I would have to...
* run npm install (wait for it to fail)
* delete the contents of node_modules (side note, you cant just run npm install again b.c it gives a false positive that everything is OK, which it is not)
* set permissions to "Everyone"
* run npm install again

I can confirm these steps work.

UPDATE:
noticed this in another issue... "**It looks like a background process is preventing npm from doing the job**"... I had Visual Studio 2017 opened, so I closed that, then npm installed and it worked.

**Comment by kenany** [comment]
Doesn't have to be "Everyone", just your current user _at the very least_. I'm not sure if there's anything that `npm` can do if your current user doesn't have permission to read/write from your current directory. Hence, I closed this as resolved since it was the fault of file permissions, not `npm` itself.

**Comment by EBMike** [comment]
> I closed this as resolved since it was the fault of file permissions, not npm itself.

@KenanY what people are going to notice is this... it worked fine with npm 4x and now it doesn't

**Comment by kenany** [comment]
That was not clear to me at the time of closing (since the OP provided very little information about their environment). We can move this to a proper bug report if this is indeed a bug with `npm@5`.

**Comment by rjgotten** [comment]
> I was able to fix this by changing the permission of the folder i was trying to run npm install or ng new from to Everyone with full control

Placebo effect, probably.

There are multiple race conditions in NPM 5.x that causes EPERM errors to be raised on Windows.
Basically everything later than 5.0.4 is affected to some degree, iirc. But as these are race conditions, it depends heavily on your exact hardware and software environment if (and how frequent) these bugs will trigger.

All - or atleast most - of these are artifacts of the fact that NPM relies on atomically writing to files. This is a unix thing, which is not at all supported well on Windows where you need to take/wait on locks.

It seems that rather than patching the root cause, there is instead a lot of patchwork happening left and right where to mitigate the fallout of this architectural problem: e.g. manually surpressing EPERM errors or trying to perform operations in such a manner that you cannot get conflicting read/write locks.
